Paginated ajax call in job_query

IsleZhu opened the pull request:
Changed the DataTables' data source in 'job_query.html' to paginated ajax data. Parsed pagination args to API.

vineetbansal commented on src/guidescanpy/flask/templates/job_query.html:
```diff
@@ -156,32 +137,69 @@ <h5 class="card-header">{{ region }}</h5>
-                        columnDefs: [
-
+                        serverSide: true,
+                        pageLength: 5,
```
maybe this can go away?

vineetbansal commented on src/guidescanpy/flask/templates/job_query.html:
```diff
@@ -156,32 +137,69 @@ <h5 class="card-header">{{ region }}</h5>
+                        ajax: {
+                            url: "{{url_for('job_query.result', format="json", job_id=result.task_id)}}",
+                            data: function (d) {
+                                d.per_page = 5;
```
let's keep default as 10

vineetbansal commented on src/guidescanpy/flask/templates/job_query.html:
```diff
@@ -156,32 +137,69 @@ <h5 class="card-header">{{ region }}</h5>
                     igv.createBrowser(igvDiv, options);
 
+                    var region = $('.resulttable').data('region');
                     var table = $('.resulttable').DataTable({
```
This will need to go in a loop for each datatable found (looking for class `.resulttable` is okay).

IsleZhu commented:
I kept the 'result' function the same as the PR #34, and added a new function 'result_dt' to generate the DataTable source data to avoid messing with the original function. I also added server-side sorting functionality to the DataTables, and set a default order of ['coordinate', 'asc'].

vineetbansal commented:
Yes - a separate `result_dt` function is a good idea since so many things in that function are specialized. However, I made some tiny tweaks in the `get_result` function so as to not pass in the `dt` parameter (in general, it's best to have generally useful functions not be aware of who's calling them, but let the caller decide if they want to slightly reformat the results or ignore certain return values of the function - for example, the non-DT caller can simply ignore `total_hits` in this case).

I've tested this out locally but you may want to do so as well before approving it.

IsleZhu commented:
Thank you, Vineet! The functions look more organized this way. I also tested it locally, it works very well. We can approve it now.
